presto-expressions/src/test/java/com/facebook/presto/expressions/translator/TestRowExpressionTranslator.java

```diff
@@ -0,0 +1,224 @@
+package com.facebook.presto.expressions.translator;
+
+import com.facebook.presto.metadata.FunctionManager;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.metadata.MetadataManager;
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.function.FunctionHandle;
+import com.facebook.presto.spi.function.FunctionMetadata;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.StandardFunctionResolution;
+import com.facebook.presto.spi.relation.CallExpression;
+import com.facebook.presto.spi.relation.ConstantExpression;
+import com.facebook.presto.spi.relation.DeterminismEvaluator;
+import com.facebook.presto.spi.relation.FullyQualifiedName;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.SpecialFormExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.sql.TestingRowExpressionTranslator;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.relational.FunctionResolution;
+import com.facebook.presto.sql.relational.RowExpressionDeterminismEvaluator;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+import org.weakref.jmx.internal.guava.collect.ImmutableSet;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.expressions.translator.FunctionTranslator.buildFunctionTranslator;
+import static com.facebook.presto.expressions.translator.RowExpressionTreeTranslator.translateWith;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.expression;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static org.testng.Assert.assertEquals;
+
+public class TestRowExpressionTranslator
+{
+    private static class TestFunctionHandle
+            implements FunctionHandle
+    {
+        @Override
+        public FullyQualifiedName.Prefix getFunctionNamespace()
+        {
+            return FullyQualifiedName.of("a.b.c").getPrefix();
+        }
+    }
+
+    private static final FunctionHandle TEST_FUNCTION = new TestFunctionHandle();
```

Comment:
nit: no need to create the test class; use lambda:
```java
private static final FunctionHandle TEST_FUNCTION = () -> FullyQualifiedName.of("a.b.c").getPrefix();

Reply:
Cool forgot you could do that :)